relevanceai/dataset_api/dataset_write.py

```diff
@@ -166,10 +170,27 @@ def insert_pandas_dataframe(self, df: pd.DataFrame, *args, **kwargs):
         """
         import pandas as pd
 
+        if col_for_id is not None:
+            df["_id"] = df[col_for_id]
+
+        else:
+            uuids = [uuid.uuid4() for _ in range(len(df))]
+            df["_id"] = uuids
+
+        def _is_valid(v):
+            try:
+                if pd.isna(v):
+                    return False
+                else:
+                    return True
+            except:
+                pass
```

Comment:
Can you explain why this is in a `try-except` block? What is the edge case here that's being considered? Otherwise this just returns `False` ?

Reply:
So, possible dtypes in a doc are array, str, int, float etc. `pd.isna(v)` only works when `v` is int/float or str not when it is array; in any case, array is always valid. I dont want to use `isinstance` here cos numpy datatypes as well